packages/record/src/components/RecordField/helpers/valueFormatter.js

```diff
@@ -12,27 +14,43 @@ export const percent = (value) => {
     }).format(value);
 };
 
+const isValidDate = (date) => {
+    return isNumber(date.getTime());
+};
+
 export const dateTime = (value) => {
-    return new Intl.DateTimeFormat(undefined, {
-        year: 'numeric',
-        month: 'numeric',
-        day: 'numeric',
-        hour: 'numeric',
-        minute: 'numeric',
-        second: 'numeric',
-    }).format(value);
+    const date = new Date(value);
+    if (isValidDate(date)) {
+        return new Intl.DateTimeFormat(undefined, {
+            year: 'numeric',
+            month: 'numeric',
+            day: 'numeric',
+            hour: 'numeric',
+            minute: 'numeric',
+            second: 'numeric',
+        }).format(date);
+    }
```

Comment:
here we should add a simple test mounting a the RecordField with ```type="dateTime"``` and passing an invalid date like a string and checks that it renders an empty string and does not crash